qadence/models/qnn.py

```diff
@@ -218,3 +222,43 @@ def _format_to_dict(self, values: Tensor) -> dict[str, Tensor]:
         assert values.size()[1] == self.in_features, msg
 
         return {var.name: values[:, self.inputs.index(var)] for var in self.inputs}
+
+    def _to_dict(self, save_params: bool = False) -> dict:
+        d = dict()
+        try:
+            d = super()._to_dict(save_params)
+            d[self.__class__.__name__]["inputs"] = [str(i) for i in self.inputs]
+            logger.debug(f"{self.__class__.__name__} serialized to {d}.")
+        except Exception as e:
+            logger.warning(f"Unable to serialize {self.__class__.__name__} due to {e}.")
+        return d
+
+    @classmethod
+    def _from_dict(cls, d: dict, as_torch: bool = False) -> QNN:
+        from qadence.serialization import deserialize
+
+        qnn: QNN
+        try:
+            qm_dict = d[cls.__name__]
+            qnn = cls(
+                circuit=QuantumCircuit._from_dict(qm_dict["circuit"]),
+                observable=[deserialize(q_obs) for q_obs in qm_dict["observable"]],  # type: ignore[misc]
+                backend=qm_dict["backend"],
+                diff_mode=qm_dict["diff_mode"],
+                measurement=Measurements._from_dict(qm_dict["measurement"]),
```

Comment:
Don't you miss the `noise` parameter ? On that note, if QNN is a QM I guess the `mitigation` parameter should be present.

Reply:
It seems noise and mitigation are not there in the serialization of `QuantumModel` as well

Reply:
added